### DISCUSSION TITLE
G722.1 codec

### DISCUSSION THREAD

**Discussion by niallgal** [baresip/baresip#2907] [Q&A]
What library should be installed for G722.1 codec?

**Comment by juha-h** [comment]
niallgal writes:

> What library should be installed for G722.1 codec?

I used the one from freeswitch.  Building debian package didn't work out
of the box, but I was able to build one after tweaking files in its
debian subdir.    I don't dare to share the source, since I think license
is needed from Polycom.

g722_1 provides good voice quality and on Android I feel it does better
job than opus.

-- Juha

**Comment by niallgal** [comment]
I would appreciate your guidance on how to get the  G722.1 codec library from the Freeswitch package.

**Comment by juha-h** [comment]
niallgal writes:

> I would appreciate your guidance on how to get the  G722.1 codec
> library from the Freeswitch package.

For example from here:

https://github.com/traviscross/freeswitch/tree/master/libs

-- Juha

**Comment by niallgal** [comment]
Many thanks.

**Comment by haricane8133** [comment]
@juha-h, @niallgal, @sreimers, could you please elaborate more on how to use G722.1 with BareSip?

I am not in an Android project

**Comment by juha-h** [comment]
Hari R writes:

> @juha-h, @niallgal, could you please elaborate more on how to use
> G722.1 with BareSip?

See https://github.com/juha-h/libbaresip-android/blob/master/Makefile
how module g722.1 can be build.  Once done, load module g7221.so and put
G7221/16000/1 in codecs list.
